sdl2promela/sdl/model.py

```diff
@@ -1451,6 +1451,54 @@ def convert(source: ogAST.Decision) -> Action:
     return None
 
 
+def _parseObserverInput(desc: str) -> Tuple[ObserverAttachmentInfo, str]:
+    info = ObserverAttachmentInfo()
+    parameter = ""
+    if desc.lower().startswith("input"):
+        info.kind = ObservedSignalKind.INPUT
+    elif desc.lower().startswith("output"):
+        info.kind = ObservedSignalKind.OUTPUT
+    else:
+        raise ValueError("Cannot parse continuous signal")
+    try:
+        if "(" in desc:
+            # If desc contains '(', then it contains parameter
+            # and needs to be parsed differently
+            open_position = desc.index("(")
+            close_position = desc.index(")", open_position + 1)
+            first_part = desc[0:open_position]
+            words = first_part.split()
+            info.originalSignalName = words[1]
+            info.observerSignalName = f"obs_{words[1]}"
+
+            parameter = desc[open_position + 1 : close_position]
+
+            second_part = desc[close_position + 1 :]
+            words = second_part.split()
+            if words[0].lower() == "from":
```

Comment:
Could we try to refactor this part a bit, to extract the to/from parsing and separate it from () handling?

Reply:
refactored